### ISSUE TITLE
local_domain_name not in compliance with Special-Use Domain 'home.arpa.'  RFC 8375

### ISSUE THREAD

**Issue by basiladdington** [AdguardTeam/AdGuardHome#4981]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to request a feature or enhancement and not ask a question

#### What problem are you trying to solve?
**Setting local search domains and allowing DHCP to populate the domain to clients.**

Currently, the setting defaults to 'lan'. (This setting is not in compliance with RFC 8375.) Casual users are not notified of this except by reading the wiki configuration documentation. My Linksys router allows me to specify a local search domain, which is then populated to the clients via the DHCP process (as verified by my MacBook Pro in the network settings). After switching from the router's DHCP to Adguard Home's DHCP the search domain was not populated even with the default 'lan'. 

#### Proposed solution

1) Be compliant with RFC8375 and change the default local_domain_name from 'lan' to 'home.arpa'. Even if you keep 'lan' the following suggestions are still valid.
2) Allow users to change 'local_domain_name' via the web admin interface.
3) Notify them of the default setting in the web interface and allow changes.
4) Have your DHCP server the local_domain_name to clients.

#### Alternatives considered
Manual updates to AdGuard Home and Clients where they available in their settings

#### Additional information
https://www.rfc-editor.org/rfc/rfc8375

   The domain name 'home.arpa.' is to be used for naming within
   residential homenets.  Names ending with '.home.arpa.' reference a
   zone that is served locally, the contents of which are unique only to
   a particular homenet and are not globally unique.  Such names refer
   to nodes and/or services that are located within a homenet (e.g., a
   printer or a toaster).

   DNS queries for names ending with '.home.arpa.' are resolved using
   local resolvers on the homenet.  Such queries MUST NOT be recursively
   forwarded to servers outside the logical boundaries of the homenet.

   Some service discovery user interfaces that are expected to be used
   on homenets conceal information such as domain names from end users.
   However, in some cases, it is still expected that users will need to
   see, remember, and even type names ending with '.home.arpa.'.  The
   Homenet Working Group hopes that this name will in some way indicate
   to as many readers as possible that such domain names are referring
   to devices in the home, but we recognize that it is an imperfect
   solution.

But compare with RFC6762

   Some network operators setting up private internal networks
   ("intranets") have used unregistered top-level domains, and some may
   have used the ".local" top-level domain.  Using ".local" as a private
   top-level domain conflicts with Multicast DNS and may cause problems
   for users.  Clients can be configured to send both Multicast and
   Unicast DNS queries in parallel for these names, and this does allow
   names to be looked up both ways, but this results in additional
   network traffic and additional delays in name resolution, as well as
   potentially creating user confusion when it is not clear whether any
   given result was received via link-local multicast from a peer on the
   same link, or from the configured unicast name server.  Because of
   this, we recommend against using ".local" as a private Unicast DNS
   top-level domain.  We do not recommend use of unregistered top-level
   domains at all, but should network operators decide to do this, the
   following top-level domains have been used on private internal
   networks without the problems caused by trying to reuse ".local." for
   this purpose:

      .intranet.
      .internal.
      .private.
      .corp.
      .home.
      .lan.

**Comment by megamorf** [comment]
I have set up Adguard Home (adguard/adguardhome:v0.107.21) in my home network today used as DHCP and DNS server.

The relevant part in the `AdGuardHome.yaml` is:

```
dhcp:
  enabled: true
  interface_name: eth0
  local_domain_name: home.arpa
  dhcpv4:
    gateway_ip: 192.168.39.1
    subnet_mask: 255.255.255.0
    range_start: 192.168.39.100
    range_end: 192.168.39.200
    lease_duration: 86400
    icmp_timeout_msec: 1000
    options: []
  dhcpv6:
    range_start: ""
    lease_duration: 86400
    ra_slaac_only: false
    ra_allow_slaac: false
clients:
  runtime_sources:
    whois: true
    arp: true
    rdns: true
    dhcp: true
    hosts: true
  persistent: []
```

However, the lookup for non-fqdn hostnames seems to fail (same behavior when tested on MacOs and Linux):

```plaintext
▶ nslookup atlas
Server:		192.168.39.2
Address:	192.168.39.2#53

** server can't find atlas: NXDOMAIN


~
▶ nslookup atlas.home.arpa
Server:		192.168.39.2
Address:	192.168.39.2#53

Non-authoritative answer:
Name:	atlas.home.arpa
Address: 192.168.39.10
```

It seems that the Adguard Home DHCP server is not correctly advertising the domain name (option 15) and the DNS search list (option 119).

**Comment by Rawi666** [comment]
I have the same issue. Even on absolutely default dhcp settings the domain is not populated and this is seen on windows client in ipconfig /all as the following 
![image](https://user-images.githubusercontent.com/5843854/230741361-29e30626-59e9-49a0-abdf-e65c2a0a4bea.png)
DHCP from openwrt correctly assigns 'lan' value in this field so I assume this is a bug in AdGuard DHCP implementation?

**Comment by CannyGent** [comment]
I have found this issue too and had to resolve by setting dhcp option 15 in the .yaml configuration file. This should ideally be an option configurable from the GUI or documented alongside the local_domain_name configuration which I expected to set option 15 or option 119.

**Comment by Ed1123** [comment]
I believe this is related: https://github.com/hassio-addons/addon-adguard-home/issues/410

Also, I wasn't able to make the options work by manually modifying the configuration file. I'm running the DHCP on a Pi4 docker.

This config is the last thing that I tried, setting the 119 option to text home.arpa.

Another issue that I have that may be related is that I can't resolve the Pi hostname at all.

Any help is appreciate it. :)

```
http:
  pprof:
    port: 6060
    enabled: false
  address: 0.0.0.0:80
  session_ttl: 720h

auth_attempts: 5
block_auth_min: 15
http_proxy: ""
language: en
theme: auto
dns:
  bind_hosts:
    - 0.0.0.0
  port: 53
  anonymize_client_ip: false
  ratelimit: 20
  ratelimit_subnet_len_ipv4: 24
  ratelimit_subnet_len_ipv6: 56
  ratelimit_whitelist: []
  refuse_any: true
  upstream_dns:
    - https://dns10.quad9.net/dns-query
    - 94.140.14.14
    - 94.140.15.15
  upstream_dns_file: ""
  bootstrap_dns:
    - 9.9.9.10
    - 149.112.112.10
    - 2620:fe::10
    - 2620:fe::fe:10
  fallback_dns: []
  upstream_mode: load_balance
  fastest_timeout: 1s
  allowed_clients: []
  disallowed_clients: []
  blocked_hosts:
    - version.bind
    - id.server
    - hostname.bind
  trusted_proxies:
    - 127.0.0.0/8
    - ::1/128
  cache_size: 4194304
  cache_ttl_min: 0
  cache_ttl_max: 0
  cache_optimistic: false
    bogus_nxdomain: []
  aaaa_disabled: false
  enable_dnssec: false
  edns_client_subnet:
    custom_ip: ""
    enabled: false
    use_custom: false
  max_goroutines: 300
  handle_ddr: true
  ipset: []
  ipset_file: ""
  bootstrap_prefer_ipv6: false
  upstream_timeout: 10s
  private_networks: []
  use_private_ptr_resolvers: false
  local_ptr_upstreams: []
  use_dns64: false
  dns64_prefixes: []
  serve_http3: false
  use_http3_upstreams: false
  serve_plain_dns: true
  hostsfile_enabled: false
tls:
  enabled: false
  server_name: ""
  force_https: false
  port_https: 443
  port_dns_over_tls: 853
  port_dns_over_quic: 853
  port_dnscrypt: 0
  dnscrypt_config_file: ""
  allow_unencrypted_doh: false
  certificate_chain: ""
  private_key: ""
  certificate_path: ""
  private_key_path: ""
  strict_sni_check: false
querylog:
  dir_path: ""
  ignored: []
  interval: 2160h
  size_memory: 1000
  enabled: true
  file_enabled: true
statistics:
  dir_path: ""
  ignored: []
   interval: 2160h
  size_memory: 1000
  enabled: true
  file_enabled: true
statistics:
  dir_path: ""
  ignored: []
  interval: 24h
  enabled: true
whitelist_filters: []
user_rules:
  - ""
dhcp:
  enabled: true
  interface_name: eth0
  local_domain_name: home.arpa
  dhcpv4:
    gateway_ip: 192.168.100.254
    subnet_mask: 255.255.255.0
    range_start: 192.168.100.100
    range_end: 192.168.100.150
    lease_duration: 86400
    icmp_timeout_msec: 1000
    options:
      - 15 text home.arpa
      - 119 text home.arpa
  dhcpv6:
    range_start: ""
    lease_duration: 86400
    ra_slaac_only: false
    ra_allow_slaac: false
filtering:
  blocking_ipv4: ""
  blocking_ipv6: ""
  blocked_services:
    schedule:
      time_zone: UTC
    ids: []
  protection_disabled_until: null
  safe_search:
  safe_search:
    enabled: false
    bing: true
    duckduckgo: true
    google: true
    pixabay: true
    yandex: true
    youtube: true
  blocking_mode: default
  parental_block_host: family-block.dns.adguard.com
  safebrowsing_block_host: standard-block.dns.adguard.com
  rewrites: []
  safebrowsing_cache_size: 1048576
  safesearch_cache_size: 1048576
  parental_cache_size: 1048576
  cache_time: 30
  filters_update_interval: 12
  blocked_response_ttl: 10
  filtering_enabled: true
  parental_enabled: false
  safebrowsing_enabled: false
  protection_enabled: true
clients:
  runtime_sources:
    whois: true
    arp: true
    rdns: true
    dhcp: true
    hosts: true
  persistent: []
log:
  file: ""
  max_backups: 0
  max_size: 100
  max_age: 3
  compress: false
  local_time: false
  verbose: false
os:
  group: ""
  user: ""
  rlimit_nofile: 0
schema_version: 28
```

```
> switch.home.arpa
Server:		192.168.100.21
Address:	192.168.100.21#53

Non-authoritative answer:
Name:	switch.home.arpa
Address: 192.168.100.103
> switch
Server:		192.168.100.21
Address:	192.168.100.21#53

** server can't find switch: NXDOMAIN
```

**Comment by pesaventofilippo** [comment]
Same issue here. Whether the local_domain_name follows RFC 8375 or not, I think AdGuard home should advertise the local domain name to DHCP clients, to allow queries for `<hostname>` rather than `<hostname>.<local_domain_name>`

Also: with the default settings queries for non-fqdn domains get forwarded upstream, which is not ideal
